tests/net/gnrc_ipv6_nib/main.c

```diff
@@ -1139,40 +1177,42 @@ static void test_handle_pkt__rtr_adv__success(uint8_t rtr_adv_flags,
                                                          &_loc_gb) >= 0,
                                 "Address was not configured by PIO");
 
-            /* Check if SLAAC generated a neighbor solicitation */
-            TEST_ASSERT_EQUAL_INT(1, msg_avail());
-            msg_receive(&msg);
-            TEST_ASSERT_EQUAL_INT(GNRC_NETAPI_MSG_TYPE_SND, msg.type);
-            pkt = msg.content.ptr;
-            /* first snip is a netif header to _mock_netif */
-            TEST_ASSERT_NOT_NULL(pkt);
-            TEST_ASSERT_EQUAL_INT(GNRC_NETTYPE_NETIF, pkt->type);
-            TEST_ASSERT(sizeof(gnrc_netif_hdr_t) <= pkt->size);
-            netif_hdr = pkt->data;
-            TEST_ASSERT_EQUAL_INT(_mock_netif->pid, netif_hdr->if_pid);
-            /* second snip is an IPv6 header to solicited nodes of _loc_gb */
-            TEST_ASSERT_NOT_NULL(pkt->next);
-            TEST_ASSERT_EQUAL_INT(GNRC_NETTYPE_IPV6, pkt->next->type);
-            TEST_ASSERT_EQUAL_INT(sizeof(ipv6_hdr_t), pkt->next->size);
-            ipv6_hdr = pkt->next->data;
-            TEST_ASSERT_MESSAGE(ipv6_addr_equal(&ipv6_hdr->dst,
-                                                &_loc_sol_nodes),
-                                "ipv6_hdr->dst != _loc_sol_nodes");
-            TEST_ASSERT_EQUAL_INT(255, ipv6_hdr->hl);
-            /* third snip is a valid solicited neighbor solicitation to
-             * _loc_gb */
-            TEST_ASSERT_NOT_NULL(pkt->next->next);
-            TEST_ASSERT_EQUAL_INT(GNRC_NETTYPE_ICMPV6, pkt->next->next->type);
-            TEST_ASSERT_EQUAL_INT(sizeof(ndp_nbr_sol_t), pkt->next->next->size);
-            nbr_sol = pkt->next->next->data;
-            TEST_ASSERT_EQUAL_INT(ICMPV6_NBR_SOL, nbr_sol->type);
-            TEST_ASSERT_EQUAL_INT(0, nbr_sol->code);
-            TEST_ASSERT(!ipv6_addr_is_multicast(&nbr_sol->tgt));
-            TEST_ASSERT_MESSAGE(ipv6_addr_equal(&_loc_gb, &nbr_sol->tgt),
-                                "_loc_gb != nbr_sol->tgt");
-            /* no further options */
-            TEST_ASSERT_NULL(pkt->next->next->next);
-            gnrc_pktbuf_release(pkt);
+            if (is_first_check) {
```

Comment:
Could you factor this out to a new function, please?
And then not call this inside `test_handle_pkt__rtr_adv__options_success()`.
Effectively it is testing if Duplicate Address Detection is triggered by SLAAC.